plugin.rb

```diff
@@ -291,23 +291,25 @@ def self.skip_db?
 
   query = <<~SQL
     WITH x AS (
-      SELECT p.user_id, COUNT(DISTINCT st.id) AS solutions
-      FROM discourse_solved_solved_topics AS st
-      JOIN posts AS p
-         ON p.id = st.answer_post_id
-        AND COALESCE(st.created_at, :since) > :since
-        AND p.deleted_at IS NULL
-      JOIN topics AS t
-         ON t.id = st.topic_id
-        AND t.archetype <> 'private_message'
-        AND t.deleted_at IS NULL
-      JOIN users AS u
-         ON u.id = p.user_id
-      WHERE u.id > 0
-        AND u.active
-        AND u.silenced_till IS NULL
-        AND u.suspended_till IS NULL
-      GROUP BY p.user_id
+      SELECT users.id AS user_id,
+             COUNT(DISTINCT st.topic_id) FILTER (WHERE
+               st.topic_id IS NOT NULL AND
+               t.id IS NOT NULL AND
+               t.archetype <> 'private_message' AND
+               t.deleted_at IS NULL AND
+               p.deleted_at IS NULL
+             ) AS solutions
+      FROM users
+      LEFT JOIN posts p ON p.user_id = users.id
+      LEFT JOIN discourse_solved_solved_topics st
+             ON st.answer_post_id = p.id
+             AND st.created_at >= :since
+      LEFT JOIN topics t ON t.id = st.topic_id
+      WHERE users.id > 0
+        AND users.active
+        AND users.silenced_till IS NULL
+        AND users.suspended_till IS NULL
+      GROUP BY users.id
```

Comment:
I have some performance concerns about the new query because it has to join the entire users table against the `posts` table. The previous query only looped through the `discourse_solved_solved_topics` which is going to have way less rows as compared to the `users` table.

To account for users with zero solutions, did you consider just setting the count for all users to zero first in an initial query before filling up the correct count using the original query?